test/aliases.js

```diff
@@ -0,0 +1,116 @@
+'use strict'
+
+const { test } = require('tap')
+const Fastify = require('fastify')
+const GQL = require('..')
+
+const schema = `
+type Query {
+  me: User
+}
+
+type Metadata {
+  info: String!
+}
+
+type User {
+  id: ID!
+  name: String!
+  quote(input: String!): String!
+  metadata(input: String!): Metadata!
+}`
+
+const users = {
+  u1: {
+    id: 'u1',
+    name: 'John'
+  },
+  u2: {
+    id: 'u2',
+    name: 'Jane'
+  }
+}
+
+const resolvers = {
+  Query: {
+    me: (root, args, context, info) => {
+      return users.u1
+    }
+  },
+  User: {
+    quote: (user, args, context, info) => {
+      return args.input
+    },
+    metadata: (user, args, context, info) => {
+      return {
+        info: args.input
+      }
+    }
+  }
+}
+
+function createTestServer (t, customResolvers = resolvers) {
+  const app = Fastify()
+  t.tearDown(app.close.bind(app))
+  app.register(GQL, { schema, resolvers: customResolvers })
+  return app
+}
+
+test('should support aliases', async t => {
+  t.plan(1)
+  const app = await createTestServer(t)
+
+  const query = `
+    query {
+      user: me {
+        id
+        name
+        newName: name
+        otherName: name
+        quote(input: "quote")
+        firstQuote: quote(input: "foo")
+        secondQuote: quote(input: "bar")
```

Comment:
I didn't even know this syntax was possible, I've learned something new today!

Reply:
Awesome!! :) Yeah I've found it to be super useful when you want multiple results from the same field in a single request!

Reply:
a little bit of shameless plug but in [gqless](https://github.com/gqless/gqless) we use them to keep track of differences in variables in fields, they are very useful

Reply:
That's really neat!